android/modules/network/src/java/ti/modules/titanium/network/TiAuthenticator.java

```diff
@@ -27,9 +29,13 @@ public PasswordAuthentication getPasswordAuthentication() {
         if (domain != null && !domain.isEmpty()) {
             user = domain + "\\" + user;
         }
-        if (password != null) {
-            return new PasswordAuthentication(user, password.toCharArray());
+        if (retryCount < MAX_RETRY_COUNT) {
```

Comment:
This looks like a workaround. Can you leave a comment in the code explaining what's going on?